### PR TITLE
feat(samples): Adding query samples

### PR DESCRIPTION
**Pull request by vinagesh**
Adding sample to call query API

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.azure.digitaltwins.core.util;
+
+import com.azure.core.http.rest.PagedResponse;
+
+import java.util.Objects;
+
+/**
+ * A helper to extract the query charge from the query response.
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java:8]: Could you add a note about what a query charge is?

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java [hunk 2]
```diff
@@ -333,4 +332,48 @@ public static void connectTwinsTogether() throws IOException, InterruptedExcepti
...
         connectTwinsLatch.await(MAX_WAIT_TIME_ASYNC_OPERATIONS_IN_SECONDS, TimeUnit.SECONDS);
     }
+
+    /**
+     *
+     * Queries for all the digital twins.
+     * @throws IOException If an I/O error is thrown when accessing the starting file.
+     * @throws InterruptedException If the current thread is interrupted while waiting to acquire latch.
+     */
+    public static void queryTwins() throws IOException, InterruptedException {
```
Comment (timtay-microsoft) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:342]: Are IOExceptions actually thrown? I don't see any of our client methods declaring it.

**Review by timtay-microsoft — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java (hunk 1 above)
Comment (vinagesh) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java:8]: sure will do

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java (hunk 2 above)
Comment (vinagesh) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:342]: Nope. IOException is not needed for query. Will remove it.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * An Azure Digital Twins Query Unit (QU) is a unit of on-demand computation that's used to execute your Azure Digital Twins queries.
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java:9]: 👍

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ */
+public final class QueryChargeHelper {
+    private static final String queryChargeHeader = "query-charge";
+
+    public static <T> Float getQueryCharge(PagedResponse<T> page) {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java:14]: should we return a primitive type `float` instead, with value 0? Or do you think that might be misleading?

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java [hunk 5]
```diff
@@ -333,4 +332,48 @@ public static void connectTwinsTogether() throws IOException, InterruptedExcepti
         // Wait until the latch count reaches zero, signifying that the async calls have completed successfully.
         connectTwinsLatch.await(MAX_WAIT_TIME_ASYNC_OPERATIONS_IN_SECONDS, TimeUnit.SECONDS);
     }
+
+    /**
+     *
+     * Queries for all the digital twins.
+     * @throws IOException If an I/O error is thrown when accessing the starting file.
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:339]: this "throws" comment can be removed

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java [hunk 6]
```diff
@@ -333,4 +332,48 @@ public static void connectTwinsTogether() throws IOException, InterruptedExcepti
...
+     * @throws InterruptedException If the current thread is interrupted while waiting to acquire latch.
+     */
+    public static void queryTwins() throws InterruptedException {
+        System.out.println("QUERY DIGITAL TWINS");
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:343]: We've been using the `ConsoleLogger` helper class for printing console statements.

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java (hunk 4 above)
Comment (vinagesh) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/util/QueryChargeHelper.java:14]: I felt 0 would be misleading compared to null so did not use the primitive type.

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java (hunk 6 above)
Comment (vinagesh) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:343]: Are you sure? I say  System.out.println all across the file.

Comment (abhipsaMisra) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:342]: That's weird, are you sure you have the latest code?

Comment (vinagesh) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:342]: Looks like it has changed after I rebased. Let me switch it.

**Pull request merged**